src/mutator.c

```diff
@@ -323,27 +325,31 @@ void afl_mutfunc_clone_bytes(afl_mutator_t *mutator, afl_input_t *input) {
 
   clone_to = afl_rand_below(rand, size);
 
-  u8 *current_bytes = input->bytes;
+  // u8 *current_bytes = input->bytes;
 
   if (actually_clone) {
 
     clone_len = choose_block_len(rand, size);
     clone_from = afl_rand_below(rand, size - clone_len + 1);
 
-    input->bytes = afl_insert_substring(input->bytes, size, input->bytes + clone_from, clone_len, clone_to);
+    mutator->mutate_buf = afl_realloc(mutator->mutate_buf, clone_len + size);
```

Comment:
Maybe we should pass a reallocable buf** typedef to the mutfunc so that it can realloc internally? 
As we cannot know the resulting buf size before

Reply:
Well I was using the current `afl_realloc` right now, it took in `*buf` type, so I store that reallocable buffer pointer in the mutator itself.  And update it. As of size, don't we calculate it for every mutation in the function itself? Although not sure.

Reply:
Aah sorry missed this comment, on holidays right now. :)
afl_realloc is fine, my point is that the mutator should maybe call it internally, after it knows how many bytes will be needed (or call it multiple times)

Reply:
Aah sorry missed this comment, on holidays right now. :)
afl_realloc is fine, my point is that the mutator should maybe call it internally, after it knows how many bytes will be needed (or call it multiple times)

Reply:
I don't completely get it. The realloc is done by the mutator only, inside the `mutate_func` :sweat_smile:  Or you mean we should move it outside `mutate_func` ?